### PR TITLE
[doctor] Warn if app is (about to be) incompatible with Play Store

### PR DESCRIPTION
**Pull request by keith-kurak**
# Why
Warn folks if they need to upgrade soon/now in order to be compatible with Play Store. Encourage upgrading to stay on top of platform requirements.

Eventually, we could do the same check for iOS the next time there's a new minimum Xcode version requirement.

# How
- New check: `StoreCompatibilityCheck`, checks one of three ways:
  - if managing own native projects: tries to check for common build.gradle configurations to see if they indicate a pre-SDK 34 version (this is very imperfect...AFAIK, we can't really evaluate build.gradle)
  - if using `expo-build-properties`: check if an older target SDK version is specified.
  - SDK version: if the other two checks don't apply, check based on Expo SDK version (<=49 supported Android SDK 33)
- Refactored `existsAndIsNotIgnoredAsync` and updated related checks and unit tests accordingly.

# Test Plan
Based on SDK version:
image

Based on expo-build-properties
image

Based on build.gradle:
image

### CODE REVIEW

**Review by betomoedano — APPROVED** [review]
LGTM 

Also tested with an old project, and works fine :) 👍👍

image

File: packages/expo-doctor/src/checks/StoreCompatibilityCheck.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      const buildGradle = fs.readFileSync(
+        path.join(projectRoot, 'android', 'build.gradle'),
+        'utf8'
+      );
+      // checks for exact string from SDK 49 bare template, will not work if it was changed
+      const targetSdkVersionRegex =
+        /^\s*targetSdkVersion\s*=\s*(?:Integer\.parseInt\(findProperty\('android\.targetSdkVersion'\)\s*\?:\s*'(\d+)'\)|'(\d+)')/m;
+      const match = buildGradle.match(targetSdkVersionRegex);
+      const targetSdkVersion = match ? parseInt(match[1], 10) : undefined;
+      if (targetSdkVersion && targetSdkVersion < 34) {
```
Comment (byCedric) [inline packages/expo-doctor/src/checks/StoreCompatibilityCheck.ts:28]: Might be nice to have the `34` listed out as constant somewhere on the top. That would be easier to update or bump.

In the future, we could also tie in "minimum versions" and their date they kick into effect. Google usually announces this ahead of time, we could incorporate that ahead of time too.

File: packages/expo-doctor/src/checks/StoreCompatibilityCheck.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      // *** CNG ***
+
+      // check if build properties overrides to a lower target SDK
+      const buildPropertiesConfig = exp.plugins?.find(
+        (plugin) => plugin[0] === 'expo-build-properties'
+      );
+      if (
+        buildPropertiesConfig &&
+        buildPropertiesConfig.length > 1 &&
+        buildPropertiesConfig[1].android.targetSdkVersion < 34
```
Comment (byCedric) [inline packages/expo-doctor/src/checks/StoreCompatibilityCheck.ts:41]: Same here, would be nice to reuse that constant

File: packages/expo-doctor/src/checks/StoreCompatibilityCheck.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      ) {
+        issue =
+          'This project is using expo-build-properties to target Android API level 33 or lower. ';
+      } else if (!semver.satisfies(exp.sdkVersion!, '>=50.0.0')) {
```
Comment (byCedric) [inline packages/expo-doctor/src/checks/StoreCompatibilityCheck.ts:45]: We also could use a constant here, or maybe even lookup the template for the current version and determine it through the template files. That way, it auto-updates itself.

**Review by byCedric — APPROVED** [review]
Left a few minor comments, and a possible future feature 😄 Nice work

**Pull request merged**